crates/plugins/src/manifest.rs

```diff
@@ -88,23 +88,32 @@ impl Architecture {
 }
 
 /// Checks whether the plugin supports the currently running version of Spin.
-pub fn check_supported_version(manifest: &PluginManifest, spin_version: &str) -> Result<()> {
+pub fn check_supported_version(
+    manifest: &PluginManifest,
+    spin_version: &str,
+    override_compatibility_check: bool,
+) -> Result<()> {
     let supported_on = &manifest.spin_compatibility;
-    inner_check_supported_version(supported_on, spin_version)
+    inner_check_supported_version(supported_on, spin_version, override_compatibility_check)
 }
 
-fn inner_check_supported_version(supported_on: &str, spin_version: &str) -> Result<()> {
+fn inner_check_supported_version(
+    supported_on: &str,
+    spin_version: &str,
+    override_compatibility_check: bool,
+) -> Result<()> {
     let comparator = VersionReq::parse(supported_on).with_context(|| {
-        format!(
-            "Could not parse manifest compatibility version {} as valid semver",
-            &supported_on,
-        )
+        format!("Could not parse manifest compatibility version {supported_on} as valid semver")
     })?;
     let version = Version::parse(spin_version)?;
     if !comparator.matches(&version) {
-        return Err(anyhow!(
-            "Spin version compatibility check failed (supported: {supported_on}, actual: {spin_version}). Try running `spin plugin update` to get latest."
+        if override_compatibility_check {
+            println!("Plugin is not compatible with this version of Spin (supported: {supported_on}, actual: {spin_version}). Check overridden ... continuing to install or execute plugin.");
```

Comment:
Suggest printing to stderr instead of stdout, particularly because we don't know if something (a script, or maybe Spin itself if we one day use stdin-stdout as an interface to plugins) is consuming stdout. 
```suggestion
            eprintln!("Plugin is not compatible with this version of Spin (supported: {supported_on}, actual: {spin_version}). Check overridden ... continuing to install or execute plugin.");
```